code/modules/mining/salvage_redemption.dm

```diff
@@ -0,0 +1,171 @@
+	. = ..()
+	update_icon(UPDATE_ICON_STATE)
+
+/obj/machinery/salvage_redemption/RefreshParts()
+	var/point_mult = SALVAGE_REDEMPTION_BASE_POINT_MULT
+	for(var/obj/item/stock_parts/component in component_parts)
+		point_mult += SALVAGE_REDEMPTION_POINT_MULT_ADD_PER_RATING * component.rating
+	// Update our values
+	point_upgrade = point_mult
+	SStgui.update_uis(src)
+
+/obj/machinery/salvage_redemption/process()
+	if(panel_open || !has_power())
+		return
+
+// Interactions
+/obj/machinery/salvage_redemption/item_interaction(mob/living/user, obj/item/used, list/modifiers)
+	if(istype(used, /obj/item/storage/part_replacer))
+		return ..()
+
+	if(!has_power() || panel_open)
+		return ..()
+
+	if(animating)
+		to_chat(user, "<span class='warning'>The machine is currently processing salvage at the moment.</span>");
+		return ITEM_INTERACT_COMPLETE
+
+	if(istype(used, /obj/item/card/id))
+		var/obj/item/card/id/ID = used
+		if(!points)
+			to_chat(user, "<span class='warning'>There are no points to claim.</span>");
+			return ITEM_INTERACT_COMPLETE
+		var/claimed = FALSE
+		for(var/access in req_access_claim)
+			if(anyone_claim || (access in ID.access))
+				ID.mining_points += points
+				ID.total_mining_points += points
+				to_chat(user, "<span class='notice'><b>[points] Salvage Points</b> claimed. You have earned a total of <b>[ID.total_mining_points] Salvage Points</b> this Shift!</span>")
+				points = 0
+				claimed = TRUE
+				break
+		if(!claimed)
+			to_chat(user, "<span class='warning'>Required access not found.</span>")
+		add_fingerprint(user)
+		return ITEM_INTERACT_COMPLETE
+
+	if(istype(used, /obj/item/storage/bag/expedition))
+		var/obj/item/storage/bag/expedition/bag = used
+		if(!bag.contents)
+			to_chat(user, "<span class='warning'>You have no salvage to redeem.</span>");
+			return ITEM_INTERACT_COMPLETE
+		for(var/obj/item/salvage/loot in bag.contents)
+			salvage_buffer |= loot
+			loot.forceMove(src)
+		process_salvage(user)
+		return ITEM_INTERACT_COMPLETE
+
+	if(istype(used, /obj/item/salvage))
+		if(used.flags & NODROP || !user.drop_item() || !used.forceMove(src))
+			to_chat(user, "<span class='warning'>[used] is stuck to your hand!</span>")
+			return ITEM_INTERACT_COMPLETE
+		var/obj/item/salvage/loot = used
+		salvage_buffer |= loot
+		process_salvage(user)
+		return ITEM_INTERACT_COMPLETE
+	return ..()
+
+/obj/machinery/salvage_redemption/crowbar_act(mob/user, obj/item/I)
+	if(default_deconstruction_crowbar(user, I))
+		return TRUE
+
+/obj/machinery/salvage_redemption/screwdriver_act(mob/user, obj/item/I)
+	if(!I.use_tool(src, user, 0, volume = 0))
+		return
+	. = TRUE
+	default_deconstruction_screwdriver(user, icon_state, icon_state, I)
+
+/obj/machinery/salvage_redemption/Destroy()
+	if(salvage_buffer)
+		for(var/obj/item/salvage/loot in salvage_buffer)
+			loot.forceMove(src.loc)
+	return ..()
```

Comment:
it looks like the code for selling salvage on the cargo shuttle hasn't been removed. so if you can redeem salvage with the SRM and get credits, and then destroy the SRM which moves all the salvage to the floor, you can double dip, right?